### PR TITLE
add openmodelica defaultexperiment as vendor specific annotations

### PR DESCRIPTION
**Pull request by arun3688**
### Related Issues

https://github.com/OpenModelica/OMSimulator/issues/750

### Purpose
This PR adds the  OpenModelica default experiment set-ups as vendor specific annotation

### changed Files

Model.cpp  -- added new vendor specific annotation  <oms::simulationInformation > <DefaultExperiment /><oms::simulationInformation >, modified the code to handle import and export
Model.h    --- added a new variable to store the signal filter

### CODE REVIEW

**Comment by lochel** [comment]
@adeas31 Does OMEdit call `oms_addSignalsToResults` or `oms_removeSignalsFromResults`?

**Comment by adeas31** [comment]
It does not but we should cover scripting as well.

**Comment by lochel** [comment]
Yeah, that's right. But not that easy to do...

**Comment by adeas31** [comment]
Wouldn't it work if signal filter is a list.

If its a lot of work then better not to expose `oms_addSignalsToResults` or `oms_removeSignalsFromResults`. Just use `oms_setSignalFilter`.

**Comment by lochel** [comment]
> If its a lot of work then better not to expose `oms_addSignalsToResults` and `oms_removeSignalsFromResults`. Just use `oms_setSignalFilter`.

That would also be my proposal for now.

**Comment by arun3688** [comment]
> > If its a lot of work then better not to expose `oms_addSignalsToResults` and `oms_removeSignalsFromResults`. Just use `oms_setSignalFilter`.
> 
> That would also be my proposal for now.

I also agree

File: src/OMSimulatorLib/OMSimulator.h [hunk 1]
```diff
@@ -99,6 +99,7 @@ OMSAPI oms_status_enu_t OMSCALL oms_getInteger(const char* cref, int* value);
 OMSAPI oms_status_enu_t OMSCALL oms_getModelState(const char* cref, oms_modelState_enu_t* modelState);
 OMSAPI oms_status_enu_t OMSCALL oms_getReal(const char* cref, double* value);
 OMSAPI oms_status_enu_t OMSCALL oms_getResultFile(const char* cref, char** filename, int* bufferSize);
+OMSAPI oms_status_enu_t OMSCALL oms_getSignalFilter(const char* cref, char** regex);
```
Comment (lochel) [inline src/OMSimulatorLib/OMSimulator.h:102]: This need to be added to the documenatation.

File: testsuite/api/test_omsExport.py [hunk 2]
```diff
@@ -33,37 +33,84 @@ def printStatus(status, expected):
...
+## 		</ssd:Annotations>
+## 	</ssd:DefaultExperiment>
 ## </ssd:SystemStructureDescription>
 ## 
 ## 0
+## warning: [oms_removeSignalsFromResults] will not update the signalFilters in ssp, use [oms_setSignalFilter]
+## status:  [correct] ok
+## warning: [oms_addSignalsToResults] will not update the signalFilters in ssp, use [oms_setSignalFilter]
 ## status:  [correct] ok
+## status:  [wrong] <?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (arun3688) [reply testsuite/api/test_omsExport.py:89]: This is due to output arguments are exchanged, it should be changed to  

`(src, status) = oms.list("model")  to  (status, src) = oms.list("model")`

File: src/OMSimulatorLib/OMSimulator.h (hunk 1 above)
Comment (arun3688) [reply src/OMSimulatorLib/OMSimulator.h:102]: added the documentation

File: testsuite/api/test_omsExport.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (arun3688) [reply testsuite/api/test_omsExport.py]: This is fixed now https://github.com/OpenModelica/OMSimulator/pull/753/commits/0f4d40d928f3a24fc4c2c2849cd8ae2f085e7e2c

[the remark this answers is not in the document]
Comment (lochel) [reply testsuite/api/test_omsExport.py]: `src` should actually be the first return value. Not sure why and when it got changed 😕

[the remark this answers is not in the document]
Comment (arun3688) [reply testsuite/api/test_omsExport.py]: changed the output arguments to be consistent with the documentation https://github.com/OpenModelica/OMSimulator/pull/753/commits/62b8f3762bfb160552e5e0e83495a254366ac6c1

**Pull request merged**